pkg/gencosmosdb/cosmosdb/database.go

```diff
@@ -27,10 +29,12 @@ type Databases struct {
 }
 
 type databaseClient struct {
+	log             *logrus.Entry
 	hc              *http.Client
 	jsonHandle      *codec.JsonHandle
 	databaseAccount string
 	masterKey       []byte
+	retry           int
```

Comment:
call this maxRetries ?
statically initialise it to 10 for now in NewDatabaseClient and see how we do?